pkg/kong/globalrules_test.go

```diff
@@ -0,0 +1,175 @@
+package kong
+
+import (
+	"testing"
+
+	"github.com/api7/kong-to-apisix/pkg/apisix"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMigrateGlobalRules(t *testing.T) {
+	var err error
+	var kongConfig Config
+	var apisixConfig apisix.Config
+
+	// KeyAuth Global Plugin Test
+	kongPluginKeyAuthConfig := make(map[string]interface{})
+	kongPluginKeyAuthConfig["key_names"] = make([]interface{}, 1)
+	kongPluginKeyAuthConfigKeyNames := make([]interface{}, 1)
+	for i, v := range []string{"apikey"} {
+		kongPluginKeyAuthConfigKeyNames[i] = v
+	}
+	kongPluginKeyAuthConfig["key_names"] = kongPluginKeyAuthConfigKeyNames
+	kongPluginKeyAuths := Plugins{
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginKeyAuth,
+			Enabled: true,
+			Config:  kongPluginKeyAuthConfig,
+		},
+	}
+	for _, kongPluginKeyAuth := range kongPluginKeyAuths {
+		kongConfig = Config{}
+		apisixConfig = apisix.Config{}
+		kongConfig.Plugins = append(kongConfig.Plugins, kongPluginKeyAuth)
+		err = MigrateGlobalRules(&kongConfig, &apisixConfig)
+		assert.NoError(t, err)
+		assert.Equal(t, kongPluginKeyAuth.ID, apisixConfig.GlobalRules[0].ID)
+	}
+
+	// RateLimiting Global Plugin Test
+	kongPluginRateLimitings := Plugins{
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"second": 5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"minute": 5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"hour":   5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"day":    5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"month":  5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+		{
+			ID:      uuid.NewV4().String(),
+			Name:    PluginRateLimiting,
+			Enabled: true,
+			Config: map[string]interface{}{
+				"year":   5,
+				"policy": PluginRateLimitingPolicyLocal,
+			},
+		},
+	}
+
+	for _, kongPluginRateLimiting := range kongPluginRateLimitings {
```

Comment:
We can improve it by providing an "expect" struct, which contains time units like "second" and expected value.

Reply:
OK, done